src/ert/scheduler/lsf_driver.py

```diff
@@ -410,6 +411,50 @@ async def _process_job_update(self, job_id: str, new_state: AnyJob) -> None:
                 del self._iens2jobid[iens]
             await self.event_queue.put(event)
 
+    async def _get_exit_code(self, job_id: str) -> int:
+        process = await asyncio.create_subprocess_exec(
+            self._bjobs_cmd,
+            "-o exit_code",
+            "-noheader",
+            job_id,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+
+        stdout, _ = await process.communicate()
+        if process.returncode:
+            return await self._get_exit_code_from_bhist(job_id)
+        else:
+            try:
+                return int(stdout.decode())
+            except ValueError:
+                # bjobs will sometimes return only "-" as exit code.
+                # running bhist will not help in this case.
+                return LSF_FAILED_JOB
+
+    async def _get_exit_code_from_bhist(self, job_id: str) -> int:
+        process = await asyncio.create_subprocess_exec(
+            self._bhist_cmd,
+            "-l",
+            "-n2",
+            job_id,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+
+        stdout, stderr = await process.communicate()
+        if process.returncode:
+            logger.warning(
+                f"bhist gave returncode {process.returncode} and error {stderr.decode()} "
```

Comment:
shall we also log `stdout` when non empty?

Reply:
I think _execute_with_retry will give you this for free

Reply:
uses `_execute_with_retry` now